aptos-indexer-processors-sdk/sdk/src/common_steps/version_tracker_step.rs

```diff
@@ -1,76 +1,129 @@
 /// if the transactions are not ordered.
-pub struct VersionTrackerStep<T, S>
+pub struct VersionTrackerStep<T, S, D>
 where
     Self: Sized + Send + 'static,
     T: Send + 'static,
-    S: ProcessorStatusSaver + Send + 'static,
+    S: ProcessorStatusSaver + Send + Sync + 'static,
+    D: DbConfig + Send + Sync + Clone + 'static,
 {
     // Last successful batch of sequentially processed transactions. Includes metadata to write to storage.
     last_success_batch: Option<TransactionContext<()>>,
     polling_interval_secs: u64,
     processor_status_saver: S,
+    processor_id: String,
+    processor_config: IndexerProcessorConfig<D>,
     _marker: PhantomData<T>,
 }
 
-impl<T, S> VersionTrackerStep<T, S>
+impl<T, S, D> VersionTrackerStep<T, S, D>
 where
     Self: Sized + Send + 'static,
     T: Send + 'static,
-    S: ProcessorStatusSaver + Send + 'static,
+    S: ProcessorStatusSaver + Send + Sync + 'static,
+    D: DbConfig + Send + Sync + Clone + 'static,
 {
-    pub fn new(processor_status_saver: S, polling_interval_secs: u64) -> Self {
+    pub fn new(
+        processor_status_saver: S,
+        polling_interval_secs: u64,
+        processor_id: &str,
+        processor_config: IndexerProcessorConfig<D>,
+    ) -> Self {
         Self {
             last_success_batch: None,
             processor_status_saver,
             polling_interval_secs,
+            processor_id: processor_id.to_string(),
+            processor_config,
             _marker: PhantomData,
         }
     }
 
     async fn save_processor_status(&mut self) -> Result<(), ProcessorError> {
         if let Some(last_success_batch) = self.last_success_batch.as_ref() {
-            self.processor_status_saver
-                .save_processor_status(last_success_batch)
-                .await
+            match self.processor_config.mode {
+                ProcessorMode::Default => {
+                    self.processor_status_saver
+                        .save_processor_status(ProcessorStatus {
+                            processor_id: self.processor_id.clone(),
+                            last_success_version: last_success_batch.metadata.end_version as i64,
+                            last_transaction_timestamp: last_success_batch
+                                .metadata
+                                .end_transaction_timestamp
+                                .as_ref()
+                                .map(|t| {
+                                    parse_timestamp(
+                                        t,
+                                        last_success_batch.metadata.end_version as i64,
+                                    )
+                                    .naive_utc()
+                                }),
+                        })
+                        .await
+                },
+                ProcessorMode::Backfill => {
+                    let backfill_config = self.processor_config.backfill_config.as_ref().unwrap();
```

Comment:
This is a big code smell, check out the alternative suggestion for the processor mode enum below.